httpx/_client.py

```diff
@@ -1043,7 +1047,10 @@ def __exit__(
         exc_value: BaseException = None,
         traceback: TracebackType = None,
     ) -> None:
-        self.close()
+        self._transport.__exit__()
+        for proxy in self._proxies.values():
+            if proxy is not None:
+                proxy.__exit__()
```

Comment:
I believe we need to send the exception information down, like this:

```suggestion
        self._transport.__exit__(exc_type, exc_value, traceback)
        for proxy in self._proxies.values():
            if proxy is not None:
                proxy.__exit__(exc_type, exc_value, traceback)
```

(I'm actually surprised the code works without passing these - HTTPCore transports allow `= None` but I'm not _sure_ this is part of the context manager protocol strictly?)